actionview/lib/action_view/helpers/date_helper.rb

```diff
@@ -91,12 +91,23 @@ module DateHelper
       #   distance_of_time_in_words(from_time, from_time + 50.minutes, scope: 'datetime.distance_in_words.short') # => "an hour"
       #   distance_of_time_in_words(from_time, from_time + 3.hours, scope: 'datetime.distance_in_words.short')    # => "3 hours"
       def distance_of_time_in_words(from_time, to_time = 0, options = {})
+        from_time = Time.at(from_time) if from_time.is_a?(Numeric)
+        to_time = Time.at(to_time) if to_time.is_a?(Numeric)
+
+        unless from_time.respond_to?(:to_time)
+          raise ArgumentError, "first argument must be convertible to_time"
+        end
+
+        unless to_time.respond_to?(:to_time)
+          raise ArgumentError, "second argument must be convertible to_time"
+        end
+
         options = {
           scope: :'datetime.distance_in_words'
         }.merge!(options)
 
-        from_time = from_time.to_time if from_time.respond_to?(:to_time)
-        to_time = to_time.to_time if to_time.respond_to?(:to_time)
+        from_time = from_time.to_time
+        to_time = to_time.to_time
```

Comment:
Seems like we could move this up to the start of the method and combine into `if` statements:

``` ruby
if from_time.is_a?(Numeric)
  from_time = Time.at(from_time)
elsif from_time.respond_to?(:to_time)
  from_time = from_time.to_time
else
  raise ArgumentError, "first argument must either be a time, date or numeric, was '#{from_time.inspect}'"
end

if to_time.is_a?(Numeric)
  to_time = Time.at(to_time)
elsif from_time.respond_to?(:to_time)
  to_time = to_time.to_time
else
  raise ArgumentError, "second argument must either be a time, date or numeric, was '#{to_time.inspect}'"
end
```